src/detect-tos.c

```diff
@@ -121,31 +121,33 @@ static DetectTosData *DetectTosParse(const char *arg, bool negate)
         goto error;
     }
 
-    const char *str_ptr;
-    res = pcre_get_substring((char *)arg, ov, MAX_SUBSTRINGS, 1,
-                             &str_ptr);
+    /* For TOS value */
+    char tosbytes_str[64] = "";
+    res = pcre_copy_substring((char *)arg, ov, MAX_SUBSTRINGS, 1,
```

Comment:
Is `tosbytes_str` guaranteed to fit 64 bytes? AFAIK, `pcre_copy_substring()` uses `memcpy()` or `strncpy()`, and both are insecure. Maybe it'll be better to initialize the variable as `{ 0 }` and use `sizeof(tosbytes_str) - 1`?
Also, I'm not sure that we still need to cast `arg` because `pcre_copy_substring()` accepts `const char *` as its first argument

Reply:
the manpage suggests that the function will return PCRE_ERROR_NOMEMORY if the size is too small, so I think we're good.